cmd/notary/keys.go

```diff
@@ -449,12 +430,11 @@ func (k *keyCommander) keysRotate(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	for _, role := range rolesToRotate {
-		if err := nRepo.RotateKey(role, k.rotateKeyServerManaged); err != nil {
-			return err
-		}
+	err = nRepo.RotateKey(rotateKeyRole, k.rotateKeyServerManaged)
+	if err == nil && k.rotateKeyServerManaged {
+		err = nRepo.Publish()
```

Comment:
I can move the call to publish to `nRepo.RotateKey` - I was thinking about adding a private `publish(changelist)` command.  RotateKey can generate an in-memory change file that gets published using that private method.